src/libDirectoryService/DSBlockPreProcessing.cpp

```diff
@@ -79,12 +65,17 @@ void DirectoryService::ComposeDSBlock(
     }
 
     uint64_t blockNum = 0;
+    BlockHash prevHash;
     uint8_t dsDifficulty = DS_POW_DIFFICULTY;
     uint8_t difficulty = POW_DIFFICULTY;
     if (m_mediator.m_dsBlockChain.GetBlockCount() > 0)
     {
         DSBlock lastBlock = m_mediator.m_dsBlockChain.GetLastBlock();
         blockNum = lastBlock.GetHeader().GetBlockNum() + 1;
+        prevHash = lastBlock.GetHeader().GetMyHash();
+
+        LOG_EPOCH(INFO, to_string(m_mediator.m_currentEpochNum).c_str(),
+                  "Prev DS block hash as per leader " << prevHash.hex());
```

Comment:
Seems "prevHash" only be used inside the scope for logging, maybe we can directly call lastBlock.GetHeader().GetMyHash() inside logging function to avoid extra variable used.

Reply:
The preHash is used to initialize the DSBlock below.

Reply:
ok, didn't notice that